### ISSUE TITLE
Train/Validation/Test Split Strategy

### ISSUE THREAD

**Issue by caesarnine** [sudarshan85/mimic3-utils#2]
@sudarshan85 I like the idea you have for maintaining approximately the same percentage of note types between the training and test set, that being said I think we may want to explore other splitting strategies.

Our data isn't really IID - it has multiple observations for a single patient, with a time element as well. Given that we want to benchmark against classification tasks downstream, we may instead want to try either a time-dependent split, or a split where there's no overlap in patients between training and test. Otherwise we may run into situations where we encounter information leakage between the training and test sets.

In practice I've found that splitting at a patient level works well: http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GroupShuffleSplit.html

**Comment by sudarshan85** [comment]
I agree with you completely. The current splitting strategy is **ONLY** for building a language model where we just want to sample all the texts regardless of dependencies.

For downstream tasks, I would think our splitting strategy would be influenced by what task we are actually trying to learn. We might have to discuss further about this to figure out what tasks we are interested in and what kind of split would be applicable. 

My guess is that once we have decided on our task and extracted the information from MIMIC required for that task (I'm thinking a big pandas DF with all the information), it would be simple enough to split them there.

**Comment by binal-patel-medamerica** [comment]
That makes sense, just wanted to make sure we were on the same page 👍 

The idea I'm generally following right now is:

1) Split all data into 80/20 train+val/test set where there's no patient overlap
2) Split train+val into 80/20 train/val where there's no patient overlap
3) Train LM on train set -> train classifier on train set -> evaluate on val set for iteration -> final eval on test set

Let's discuss further this week or next, all we really need is reproducibility in the splitting so we can benchmark against each other.

**Comment by caesarnine** [comment]
That makes sense, just wanted to make sure we were on the same page 👍 

The idea I'm generally following right now is:

1) Split all data into 80/20 train+val/test set where there's no patient overlap
2) Split train+val into 80/20 train/val where there's no patient overlap
3) Train LM on train set -> train classifier on train set -> evaluate on val set for iteration -> final eval on test set

Let's discuss further this week or next, all we really need is reproducibility in the splitting so we can benchmark against each other.
